Model/Api/UpdateCart.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Webapi\Exception as WebApiException;
+use Bolt\Boltpay\Api\UpdateCartInterface;
+use Bolt\Boltpay\Model\Api\UpdateCartCommon;
+use Bolt\Boltpay\Model\Api\UpdateCartContext;
+use Bolt\Boltpay\Model\ErrorResponse as BoltErrorResponse;
+use Bolt\Boltpay\Api\Data\CartDataInterfaceFactory;
+use Bolt\Boltpay\Api\Data\UpdateCartResultInterfaceFactory;
+use Bolt\Boltpay\Helper\Session as SessionHelper;
+
+/**
+ * Class UpdateCart
+ * 
+ * @package Bolt\Boltpay\Model\Api
+ */
+class UpdateCart extends UpdateCartCommon implements UpdateCartInterface
```

Comment:
What's you idea about class `UpdateCartCommon`? In which case you want to reuse it?

Reply:
Oh, I see in next PR
So 
```
class UpdateCart extends UpdateCartCommon implements UpdateCartInterface
class DiscountCodeValidation extends UpdateCartCommon implements DiscountCodeValidationInterface
```

Yes, this looks reasonable for me

Reply:
@vitaliyreznikov Initially the plan is to do refactoring on DiscountCodeValidation.php, so we can reuse some common functions. But since we wont touch these code in DiscountCodeValidation.php now, maybe there is no need to create such a  UpdateCartCommon class. What's your opinion?

Reply:
@vitaliyreznikov As for me, refactoring you offered looks very good. I think we can implement it, why not?

Reply:
Do you have any concerns about the refactoring?

Reply:
OK, lets keep it.